cdap-ui/app/cdap/components/AbstractWidget/Comment/CommentBox.tsx

```diff
@@ -0,0 +1,164 @@
+import Card from '@material-ui/core/Card';
+import CardHeader from '@material-ui/core/CardHeader';
+import CardContent from '@material-ui/core/CardContent';
+import makeStyle from '@material-ui/core/styles/makeStyles';
+import AccountCircleOutlinedIcon from '@material-ui/icons/AccountCircleOutlined';
+import CardActions from '@material-ui/core/CardActions';
+import Button from '@material-ui/core/Button';
+import { Theme } from '@material-ui/core/styles/createMuiTheme';
+import { COMMENT_WIDTH, IComment } from 'components/AbstractWidget/Comment/CommentConstants';
+import TextareaAutosize from '@material-ui/core/TextareaAutosize';
+import CommentMenu from 'components/AbstractWidget/Comment/CommentMenu';
+import { humanReadableDate } from 'services/helpers';
+import Markdown from 'components/Markdown';
+
+const useStyles = makeStyle<Theme, ICommentStyleProps>((theme) => {
+  return {
+    root: {
+      zIndex: 1,
+      cursor: 'text',
+      userSelect: 'text',
+    },
+    contentRoot: {
+      paddingTop: ({ editMode }) => (!editMode ? 0 : theme.spacing(2)),
+    },
+    cardAction: {
+      display: 'inline-flex',
+    },
+    textarea: {
+      width: '100%',
+      resize: 'vertical',
+      maxHeight: COMMENT_WIDTH, // Allow to resize vertically upto 300px
+      padding: theme.spacing(2),
+    },
+  };
+});
+interface ICommentStyleProps {
+  editMode: boolean;
+}
+
+interface ICommentBoxProps {
+  comment?: IComment;
+  onSave?: (comment?: IComment) => void;
+  onDelete?: () => void;
+  focus?: boolean;
+  disabled?: boolean;
+}
+
+export default function CommentBox({
+  comment,
+  onSave,
+  onDelete,
+  focus = false,
+  disabled = false,
+}: ICommentBoxProps) {
+  const isThereExistingComment = comment.content && comment.content.length > 0;
+  const [editMode, setEditMode] = React.useState(!isThereExistingComment);
+  const [localComment, setLocalComment] = React.useState(comment);
+
+  const classes = useStyles({ editMode });
+
+  const onCommentChange = (value: string) => {
+    if (!localComment.content || !localComment.content.length) {
+      setLocalComment({
+        ...comment,
+        content: value,
+        createDate: Date.now(),
+      });
+    } else {
+      setLocalComment({
+        ...comment,
+        content: value,
+      });
+    }
+  };
+  const onSaveComment = () => {
+    onSave({
+      ...comment,
+      content: localComment.content,
+    });
+  };
+  const onResetComment = () => {
+    onSave(comment);
+  };
+
+  React.useEffect(() => {
+    setLocalComment(comment);
+    if (!comment.content.length) {
+      setEditMode(true);
+    }
+  }, [comment]);
+
+  const CommentMenuLocal = disabled ? null : (
+    <CommentMenu onEdit={() => setEditMode(true)} onDelete={onDelete} />
+  );
+  if (!editMode) {
+    return (
+      <Card className={classes.root}>
+        <CardHeader
+          avatar={<AccountCircleOutlinedIcon />}
+          action={CommentMenuLocal}
+          title={'Comment'}
```

Comment:
nit: `title="Comment"`

Reply:
BTW this is gone in the next PR that is coming. We don't the title here.